src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -248,8 +264,9 @@ object VariantSampleMatrix {
     val sampleInfo = getAndCastJSON[JArray]("sample_annotations")
       .arr
       .map {
-        case JObject(List(("id", JString(id)), ("annotation", jv: JValue))) =>
-          (id, JSONAnnotationImpex.importAnnotation(jv, saSignature, "sample_annotations"))
+        case JObject(List(("id", id), ("annotation", jv))) =>
+          (JSONAnnotationImpex.importAnnotation(id, sSignature, "sample_annotations.id"),
```

Comment:
No, they come from the "sample_annotations" field of the metadata JSON.  With this change, I tried not to rename anything related to sample or variant to avoid too much chaos.  The only exception is rowkey_schem and colkey_schema on python VariantDataset since variant_schema and sample_schema were already taken.